randovania/games/am2r/exporter/patch_data_factory.py

```diff
@@ -8,17 +9,100 @@
 from randovania.game_description.assignment import PickupTarget
 from randovania.games.am2r.exporter.hint_namer import AM2RHintNamer
 from randovania.games.am2r.exporter.joke_hints import JOKE_HINTS
+from randovania.games.am2r.layout.am2r_cosmetic_patches import AM2RCosmeticPatches, MusicMode
 from randovania.games.am2r.layout.hint_configuration import ItemHintMode
 from randovania.games.game import RandovaniaGame
 from randovania.generator.pickup_pool import pickup_creator
-from randovania.lib import json_lib
+from randovania.lib import json_lib, random_lib
 
 if TYPE_CHECKING:
-    from random import Random
-
     from randovania.exporter.pickup_exporter import ExportedPickupDetails
     from randovania.games.am2r.layout.am2r_configuration import AM2RConfiguration
-    from randovania.games.am2r.layout.am2r_cosmetic_patches import AM2RCosmeticPatches
+
+
+def _construct_music_shuffle_dict(music_mode: MusicMode, rng: Random) -> dict[str, str]:
+    combat_list = [
+        "musalphafight",
+        "musancientguardian",
+        "musarachnus",
+        "museris",
+        "musgammafight",
+        "musgenesis",
+        "musomegafight",
+        "musqueen",
+        "musqueen2",
+        "musqueen3",
+        "musreactor",
+        "mustorizoa",
+        "mustorizob",
+        "muszetafight",
+        "mustester",
+    ]
+
+    exploration_list = [
+        "musarea1a",
+        "musarea1b",
+        "musarea2a",
+        "musarea2b",
+        "musarea3a",
+        "musarea3b",
+        "musarea4a",
+        "musarea4b",
+        "musarea5a",
+        "musarea5b",
+        "musarea6a",
+        "musarea7a",
+        "musarea7c",
+        "musarea8",
+        "muscaveambience",
+        "muscaveambiencea4",
+        "mushatchling",
+        "musitemamb",
+        "musitemamb2",
+        "muslabambience",
+        "musmaincave",
+        "musmaincave2",
+        "mustitle",
+    ]
+
+    fanfare_list = [
+        "musarea7b",
+        "musfanfare",
+        "musitemget",
+        "musmetroidappear",
+        "musqueenbreak",
+        "musqueenintro",
+    ]
+
+    excluded_list = [
+        "musarea7d",
+        "muscredits",
+        "musending",
+        "musintroseq",
+    ]
+
+    if music_mode == MusicMode.VANILLA:
+        return {}
+
+    # Music is now either TYPE or FULL
+    assert music_mode in (MusicMode.TYPE, MusicMode.FULL)
+
+    total_orig = combat_list + exploration_list + fanfare_list
+
+    if music_mode == MusicMode.FULL:
+        total_orig += excluded_list
+        total_new = random_lib.shuffle(rng, total_orig)
+    else:
+        # MusicMode is type
+        shuffled_combat = combat_list.copy()
+        shuffled_exploration = exploration_list.copy()
+        shuffled_fanfare = fanfare_list.copy()
```

Comment:
It probably doesn't matter that much but if I read that right, the original lists are not required anymore and skip making copies would be faster.